### PR TITLE
implementing POM and improving locators

### PR DESCRIPTION
**Pull request by ana-os-mo**
Implementation of page object model in the buy-shirt test

### CODE REVIEW

File: cypress/e2e/buy-shirt.cy.ts [hunk 1]
```diff
@@ -1,33 +1,43 @@
...
+    addressStepPage.goToShippingMenu();
+
+    shippingStepPage.markTermsOfServiceCheckbox();
+    shippingStepPage.goToPaymentMethodMenu();
+
+    paymentStepPage.selectPaymentMethod();
+    paymentStepPage.goToConfirmOrderMenu();
 
+    // assertion
     cy.get("#center_column > div > p > strong")
```
Comment (manuelq12) [inline cypress/e2e/buy-shirt.cy.ts]:
```suggestion
    paymentStepPage.getOrderMessage()
```

File: cypress/page/address-step.page.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class AddressStepPage {
+  private shippingMenu: string;
+
+  constructor() {
+    this.shippingMenu = ".cart_navigation > .button > span";
```
Comment (manuelq12) [inline cypress/page/address-step.page.ts]: This selector can be improved

File: cypress/page/login.page.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class LoginPage {
+  private emailField: string;
+  private passwordField: string;
+  private logInMenu: string;
+
+  constructor() {
+    this.emailField = "#email";
+    this.passwordField = "#passwd";
+    this.logInMenu = "#SubmitLogin > span";
```
Comment (manuelq12) [inline cypress/page/login.page.ts]: This selector can be improved

File: cypress/page/menu-content.page.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class MenuContentPage {
+  private tShirtMenu: string;
+  private menuContentPageURL: string;
+
+  constructor() {
+    this.menuContentPageURL = "http://automationpractice.com/";
+    this.tShirtMenu = ".sf-menu > :nth-child(3) > a";
```
Comment (manuelq12) [inline cypress/page/menu-content.page.ts]: This selector can be improved

File: cypress/page/payment-step.page.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+class PaymentStepPage {
+  private paymentMethod: string;
+  private confirmOrderMenu: string;
+
+  constructor() {
+    this.paymentMethod = ".bankwire > span";
+    this.confirmOrderMenu = "#cart_navigation > .button > span";
```
Comment (manuelq12) [inline cypress/page/payment-step.page.ts]: This selectors can be improved (validate if you require the span)

File: cypress/page/product-list.page.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+class ProductListPage {
+  private addToCartB: string;
+
+  constructor() {
+    this.addToCartB = ".ajax_add_to_cart_button > span";
```
Comment (manuelq12) [inline cypress/page/product-list.page.ts]: No need for span

File: cypress/page/shipping-step.page.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+class ShippingStepPage {
+  private termsOfServiceCheckbox: string;
+  private paymentMethodMenu: string;
+
+  constructor() {
+    this.termsOfServiceCheckbox = "#cgv";
+    this.paymentMethodMenu = ".cart_navigation > .button > span";
```
Comment (manuelq12) [inline cypress/page/shipping-step.page.ts]: This can be improved

File: cypress/page/shopping-cart.page.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
+class ShoppingCartPage {
+  private shoppingCartMenu: string;
+  private authenticationMenu: string;
+
+  constructor() {
+    this.shoppingCartMenu = ".button-container > .button-medium > span";
+    this.authenticationMenu = ".cart_navigation > .button > span";
```
Comment (manuelq12) [inline cypress/page/shopping-cart.page.ts]: No need for span

**Review by manuelq12 — CHANGES_REQUESTED** [review]

File: cypress/e2e/buy-shirt.cy.ts [hunk 9]
```diff
@@ -1,34 +1,44 @@
...
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    productListPage.addProductToCart();
+
+    shoppingCartPage.goToShoppingCartMenu();
+    shoppingCartPage.goToAuthenticationMenu();
+
+    loginPage.fillIdFields(userId.email, userId.password);
+    loginPage.goToUserMenu();
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts]: this can be just one method called login or signIn

File: cypress/e2e/buy-shirt.cy.ts [hunk 10]
```diff
@@ -1,34 +1,44 @@
...
+    addressStepPage.goToShippingMenu();
+
+    shippingStepPage.markTermsOfServiceCheckbox();
+    shippingStepPage.goToPaymentMethodMenu();
+
+    paymentStepPage.selectPaymentMethod();
+    paymentStepPage.goToConfirmOrderMenu();
+
+    // assertion
+    paymentStepPage.getOrderMessage(validationMessage);
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts]: the name of this method can be validateOrderMessage or if it is a "get" then return the message and perform the validation in this line

Comment (ana-os-mo) [reply cypress/e2e/buy-shirt.cy.ts]: Done. Thanks.

File: cypress/e2e/buy-shirt.cy.ts (hunk 9 above)
Comment (ana-os-mo) [reply cypress/e2e/buy-shirt.cy.ts]: done.

File: cypress/e2e/buy-shirt.cy.ts (hunk 1 above)
Comment (ana-os-mo) [reply cypress/e2e/buy-shirt.cy.ts]: done

File: cypress/page/address-step.page.ts (hunk 2 above)
Comment (ana-os-mo) [reply cypress/page/address-step.page.ts]: done

File: cypress/page/menu-content.page.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
+class MenuContentPage {
+  private tShirtMenu: string;
+  private menuContentPageURL: string;
+
+  constructor() {
+    this.menuContentPageURL = "http://automationpractice.com/";
+    this.tShirtMenu = "#block_top_menu > ul > li:nth-child(3) > a";
```
Comment (holgiosalos) [inline cypress/page/menu-content.page.ts]: as general rule, depend on the position of an element, is a bad idea for a locator, you can improve this one :)

**Review by JuanDelahozMiranda — APPROVED** [review]

**Review by manuelq12 — APPROVED** [review]

**Pull request merged**